packages/core/src/checkout-buttons/strategies/amazon-pay-v2/amazon-pay-v2-button-options.ts

```diff
@@ -1,6 +1,22 @@
+import { BuyNowCartRequestBody } from '../../../cart';
 import { AmazonPayV2ButtonParameters } from '../../../payment/strategies/amazon-pay-v2';
 
+export function isWithBuyNowFeatures(
+    options?: AmazonPayV2ButtonInitializeOptions,
+): options is WithBuyNowFeature {
+    return !!(options as WithBuyNowFeature)?.buyNowInitializeOptions;
```

Comment:
We should be avoiding the use of `as` keyword and looking at stronger typing.

Reply:
Why should we avoid this? It is acceptable use it with type guard

Reply:
Because `as` itself even in a typeguard can lead to false assertions. Same as saying a `test as object` which implies test is an object 🤔